File: src/main/java/t4/ReadMap2.java
package t4;

/**
 * Created by apple on 17/4/20.
 * <p>
 * 参数:a_b_c_d_e_f_g
 * 返回:a b c d e f g
 * 读取Map
 */
public class ReadMap2 {

    public static void main(String[] args) {
        for (String arg : args) {
            String[] splits = arg.split("_");
            for (String split : splits) {
                System.out.print(split + split.length() + "");
                System.out.print(" ");
            }

        }

    }
}

File: src/main/java/t4/ReadMap2.kt
/**
 * Created by apple on 17/4/20.
 *
 * 参数:a_b_c_d_e_f_g
 * 返回:a b c d e f g
 * 集合扁平化 flatMap
 * flatMap 是 map 的一个加强版，返回的仍然是开一个可迭代的集合
 */

// vararg 可变参数 跟一般的参数用起来差不多
// map foreach 的意思

fun main(vararg args: String) {

    args.flatMap {
        it.split("_")
    }.map {
        //字符串模版
        print("$it${it.length} ")
    }

    println();

    args.flatMap { it.split("_").map { print("$it "); } }


}